src/main/java/org/commcare/formplayer/application/MenuController.java

```diff
@@ -162,27 +162,10 @@ public BaseResponseBean navigateSessionWithAuth(@RequestBody SessionNavigationBe
     public BaseResponseBean navigateToEndpoint(@RequestBody SessionNavigationBean sessionNavigationBean,
                                                     @CookieValue(Constants.POSTGRES_DJANGO_SESSION_ID) String authToken,
                                                     HttpServletRequest request) throws Exception {
-        String endpointId = sessionNavigationBean.getEndpointId();
         MenuSession menuSession = getMenuSessionFromBean(sessionNavigationBean);
-        Endpoint endpoint = menuSession.getEndpoint(endpointId);
-        if (endpoint == null) {
-            throw new Exception("No endpoint provided to navigateToEndpoint");
-        }
-
-        SessionWrapper sessionWrapper = menuSession.getSessionWrapper();
-        EvaluationContext evalContext = sessionWrapper.getEvaluationContext();
-        HashMap<String, String> endpointArgs = new HashMap<>();
-        for (Map.Entry<String, String> arg : sessionNavigationBean.getEndpointArgs().entrySet()) {
-            endpointArgs.put(arg.getKey(), arg.getValue());
-        }
-        Endpoint.populateEndpointArgumentsToEvaluationContext(endpoint, endpointArgs, evalContext);
-
-        sessionWrapper.executeStackOperations(endpoint.getStackOperations(), evalContext);
-        runnerService.rebuildMenuSession(menuSession);
-        sessionNavigationBean.setSelections(menuSession.getSelections());
```

Comment:
Was it intentional to drop this line? The response now has `selections` as null, so web apps can't create breadcrumbs. Maybe `advanceSessionWithSelections` never sets selections because it assumes that they're already set properly?

Reply:
Ahh I removed it because it looked repetitive to me as the `advanceSessionWithSelections` should itself be enough of setting all necessary params. Though it looks like the `setSelections` call in it is under this [condition](https://github.com/dimagi/formplayer/blob/master/src/main/java/org/commcare/formplayer/services/MenuSessionRunnerService.java#L286).My sense is that we should always be setting `selections` without this if condition specially if it's used in setting the breadcrumbs. Although there seems to be a separate `breadcrumbs` array in response which looks like the entity that should ideally be responsible `breadcrumbs` ? I am quite unsure what do we use `selections` for on Web Apps.

Reply:
Ah, removing that conditional sounds good to me. Web Apps needs both breadcrumbs and selections: breadcrumbs has the text for display, and then selections has the indexes so we can navigate when someone clicks on one of the breadcrumbs.

Reply:
Thanks added https://github.com/dimagi/formplayer/pull/984/commits/00fb686bc3e99e33d53d03012d761adb4ed3fd3e